spring-cloud/spring-cloud-openfeign/src/main/java/com/baeldung/cloud/openfeign/service/impl/JSONPlaceHolderServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package com.baeldung.cloud.openfeign.service.impl;
+
+import com.baeldung.cloud.openfeign.client.JSONPlaceHolderClient;
+import com.baeldung.cloud.openfeign.model.Post;
+import com.baeldung.cloud.openfeign.service.JSONPlaceHolderService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class JSONPlaceHolderServiceImpl implements JSONPlaceHolderService {
+
+    @Autowired
+    private JSONPlaceHolderClient jsonPlaceHolderClient;
+
+    @Override
+    public List<Post> getPosts() {
+        return jsonPlaceHolderClient.getPosts();
+    }
+
+    @Override
+    public List<Post> getPostsWrong() {
+        return jsonPlaceHolderClient.getPostsWrong();
```

Comment:
It's common to have each function of a DAO or low-level client reflected by a service. When that's the case, it is usually caused by the service not adding any business logic, and merely being a conduit to the lower level client.

I think here we have an example of a `getPostsWrong` service method that's purely there to test the error handling behaviour of a client. Is this the best way to make the point? Do we want to just write the unit test that shows the error handling by testing the client directly, rather than via a service?

Just a suggestion.